es_ES/widgets-intro.md

```diff
@@ -0,0 +1,799 @@
+widgets especiales que saben como generar objetos State, que luego se usan
+para mantener el estado. Considera este ejemplo básico, usando el
+[`RaisedButton`](https://docs.flutter.io/flutter/material/RaisedButton-class.html)
+mencionado anteriormente:
+
+```dart
+class Counter extends StatefulWidget {
+  // Esta clase es la configuración para el estado. Tiene
+  // los valores (en nada) proporcionado por el padre y utilizado por el método
+  // de construcción del Estado. Los campos en una subclase Widget siempre se marcan como "final".
+
+  @override
+  _CounterState createState() => _CounterState();
+}
+
+class _CounterState extends State<Counter> {
+  int _counter = 0;
+
+  void _increment() {
+    setState(() {
+      // Esta llamada a setState le dice al framework Flutter que
+      // algo ha cambiado en este Estado, lo que hace que se vuelva a ejecutar
+      // el método de construcción siguiente para que la pantalla pueda reflejar los
+      // valores actualizados. Si cambiamos _counter sin llamar
+      // setState(), entonces el método de construcción no sería llamado de nuevo,
+      // así que no parece que pase nada.
+      _counter++;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    // Este método se vuelve a ejecutar cada vez que se llama setState, por ejemplo
+    // como se hace con el método _increment anterior.
+    // El framework Flutter ha sido optimizado para hacer que los métodos de construcción
+    // se vuelvan a ejecutar rápidamente, para que puedas reconstruir cualquier cosa que
+    // necesite ser actualizada en lugar de tener que cambiar individualmente
+    // las instancias de los widgets.
+    return Row(
+      children: <Widget>[
+        RaisedButton(
+          onPressed: _increment,
+          child: Text('Increment'),
+        ),
+        Text('Count: $_counter'),
+      ],
+    );
+  }
+}
+```
+
+Puede que se pregunte por qué StatefulWidget y State son objetos separados. En
+Flutter, estos dos tipos de objetos tienen diferentes ciclos de vida. Los Widgets son
+objetos temporales, usado para construir una presentación de la aplicación en su
+estado actual. Por otro lado, los objetos de estado son persistentes entre llamadas a
+[`build()`](https://docs.flutter.io/flutter/widgets/State/build.html),
+permitiéndoles recordar información.
+
+El ejemplo anterior acepta la entrada del usuario y utiliza directamente el resultado
+en su método de construcción.  En aplicaciones más complejas, diferentes partes de la jerarquía
+de widgets podrían ser responsables de diferentes responsabilidades; por ejemplo, un
+widget puede presentar una interfaz de usuario compleja con el objetivo de recopilar
+información específica, como una fecha o lugar, mientras que otro widget podría
+usar esa información para cambiar la presentación general.
+
+En Flutter, las notificaciones de cambio fluyen "arriba" en la jerarquía de widgets
```

Comment:
In Flutter, change notifications flow “up” the widget hierarchy by way of callbacks, while current state flows “down” to the stateless widgets that do presentation.

En Flutter, los cambios se notifican "hacia arriba" por la jerarquía de widgets mediante el uso de callbacks, mientras que el estado actual fluye "hacia abajo" a los stateless widgets que hacen la presentación

Reply:
Listo corregido